plugins/modules/aci_igmp_interface_policy.py

```diff
@@ -0,0 +1,383 @@
+    argument_spec.update(
+        name=dict(type="str"),
+        tenant=dict(type="str"),
+        description=dict(type="str"),
+        group_timeout=dict(type="int"),
+        query_interval=dict(type="int"),
+        query_response_interval=dict(type="int"),
+        last_member_count=dict(type="int"),
+        last_member_response=dict(type="int"),
+        startup_query_count=dict(type="int"),
+        startup_query_interval=dict(type="int"),
+        querier_timeout=dict(type="int"),
+        robustness_variable=dict(type="int"),
+        igmp_version=dict(type="str", choices=["v2", "v3"]),
+        allow_v3_asm=dict(type="bool"),
+        fast_leave=dict(type="bool"),
+        report_link_local_groups=dict(type="bool"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["name"]],
+            ["state", "absent", ["name"]],
+        ],
+        required_together=[
+            ["allow_v3_asm", "fast_leave", "report_link_local_groups"],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    name = module.params.get("name")
+    tenant = module.params.get("tenant")
+    description = module.params.get("description")
+    group_timeout = module.params.get("group_timeout")
+    query_interval = module.params.get("query_interval")
+    query_response_interval = module.params.get("query_response_interval")
+    last_member_count = module.params.get("last_member_count")
+    last_member_response = module.params.get("last_member_response")
+    startup_query_count = module.params.get("startup_query_count")
+    startup_query_interval = module.params.get("startup_query_interval")
+    querier_timeout = module.params.get("querier_timeout")
+    robustness_variable = module.params.get("robustness_variable")
+    igmp_version = module.params.get("igmp_version")
+    allow_v3_asm = module.params.get("allow_v3_asm")
+    fast_leave = module.params.get("fast_leave")
+    report_link_local_groups = module.params.get("report_link_local_groups")
+    state = module.params.get("state")
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class='igmpIfPol',
+            aci_rn='igmpIfPol-{0}'.format(name),
+            module_object=name,
+            target_filter={'name': name},
+        ),
+    )
+    aci.get_existing()
+
+    if state == "present":
+        if allow_v3_asm is not None:
+            if_ctrl_list = []
+            if allow_v3_asm:
+                if_ctrl_list.append("allow-v3-asm")
+            if fast_leave:
+                if_ctrl_list.append("fast-leave")
+            if report_link_local_groups:
```

Comment:
can you add test case for report_link_local_groups?

Reply:
Updated tests to include setting report_link_local_groups